### PR TITLE
feat(backend): setup hyperswarm

### PR DESCRIPTION
**Pull request by Mogge**
## 🍰 Pullrequest

Use DHT hyperswarm when DHT_TOPIC is set in .env

### Issues
- relates #1784 

### Todo

- [x] Get it running

### CODE REVIEW

**Comment by Mogge** [comment]
For some reason the topic generated by the prove of concept does not match the topic generated by this PR :(

File: backend/src/federation/index.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+          successfulRequests.push(remotePubKey)
+        })
+        // pipe it somewhere like any duplex stream
+        // process.stdin.pipe(noiseSocket).pipe(process.stdout)
+      })
+    }, POLLTIME)
+  } catch (err) {
+    console.log(err)
+  }
+}
```
Comment (clauspeterhuebner) [inline backend/src/federation/index.ts:122]: it seems to be clear logically, but its hard to give a statement about correctness or functionality with a unknown framework. So we will see and research by using it.

File: backend/src/server/createServer.ts [hunk 2]
```diff
@@ -52,6 +55,11 @@ const createServer = async (
     throw new Error('Fatal: Database Version incorrect')
   }
 
+  // start DHT hyperswarm when DHT_TOPIC is set in .env
+  if (CONFIG.DHT_TOPIC) {
+    await startDHT(con, CONFIG.DHT_TOPIC)
+  }
+
```
Comment (clauspeterhuebner) [inline backend/src/server/createServer.ts:62]: As i understand this here, the hyperswarm will be in the first steps part of our apollo-server.
In the future we should extract it as a separated modul, which will communicate with the existing moduls, right?

File: deployment/bare_metal/.env.dist [hunk 3]
```diff
@@ -26,7 +26,7 @@ COMMUNITY_REDEEM_CONTRIBUTION_URL=https://stage1.gradido.net/redeem/CL-{code}
 COMMUNITY_DESCRIPTION="Gradido Development Stage1 Test Community"
 
 # backend
-BACKEND_CONFIG_VERSION=v10.2022-09-20
+CONFIG_VERSION=v11.2022-10-27
```
Comment (clauspeterhuebner) [inline deployment/bare_metal/.env.dist:29]: was the replaced BACKEND_CONFIG_VERSION-property never used in the baremetal-scripts?
If it is in use, where are the changes about this exchange to CONFIG_VERSION?

**Review by clauspeterhuebner — APPROVED** [review]
Some open questions, but i approve it to go further.

File: deployment/bare_metal/.env.dist (hunk 3 above)
Comment (Mogge) [reply deployment/bare_metal/.env.dist:29]: Good catch. Must be `BACKEND_CONFIG_VERSION` here. As in the `backend/.env.template` it states:
`CONFIG_VERSION=$BACKEND_CONFIG_VERSION`

**Review by ulfgebhardt — APPROVED** [review]

**Pull request merged**